### PR TITLE
massive update and fix to metal3 charts

### PR DESCRIPTION
**Pull request by dbw7**
This is a fix of the majority of the issues with the metal3 charts. Not every issue has been fixed but every issue that I've been able to replicate thus far has been fixed. This includes [issue 9](https://github.com/suse-edge/charts/issues/9), [issue 14](https://github.com/suse-edge/charts/issues/14), and [issue 12](https://github.com/suse-edge/charts/issues/12).

My understanding is that [issue 10](https://github.com/suse-edge/charts/issues/10) is fixed by the other fixes but I'm not 100% certain as I haven't been able to replicate this issue myself.

There are still more issues to be fixed and tested but for now update fixes most issues.

### CODE REVIEW

File: charts/metal3-deploy/0.1.0/Chart.yaml [hunk 1]
```diff
@@ -1,27 +1,54 @@
...
   name: mariadb
   repository: file://./charts/mariadb
+  version: 0.1.1
+  
 - alias: metal3-media
   condition: global.enable_metal3_media_server
   name: media
   repository: file://./charts/media
+  version: 0.1.0
+
```
Comment (e-minguez) [inline charts/metal3-deploy/0.1.0/Chart.yaml:32]: Nit: Remove too many blank spaces

File: charts/metal3-deploy/0.1.0/charts/ironic/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+**How to Enable TLS
+
+Currently TLS is implemented only for ironic helm chart. 
+
+In order to enable TLS, common options to be maintained in values.yaml or the override files
+  ```
+  global.enable_tls: true
+  global.enable_ironic: true
+  ingress.tlsSource: self ( Valid Options: "self, letsEncrypt, secrets")
+  tls: ingress (Where to offload the TLS/SSL encryption – Valid Options: "ingress, ironic"
```
Comment (e-minguez) [inline charts/metal3-deploy/0.1.0/charts/ironic/README.md:10]:
```suggestion
tls: ingress (Where to offload the TLS/SSL encryption – Valid Options: "ingress, ironic")
```

File: charts/metal3-deploy/0.1.0/charts/ironic/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+**How to Enable TLS
+
+Currently TLS is implemented only for ironic helm chart. 
+
+In order to enable TLS, common options to be maintained in values.yaml or the override files
+  ```
+  global.enable_tls: true
```
Comment (e-minguez) [inline charts/metal3-deploy/0.1.0/charts/ironic/README.md:7]:
```suggestion
global.enable_tls: true
```

File: charts/metal3-deploy/0.1.0/charts/ironic/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  ```
+
+Additional options if 
+
+- tlsSource letsEncrypt
+
+*Pre-requistes*
+
+    • Valid Domain registered in CloudFlare
+    • Valid cloudflare API token available
```
Comment (e-minguez) [inline charts/metal3-deploy/0.1.0/charts/ironic/README.md:20]:
```suggestion
* Valid CloudFlare API token available
```

File: charts/metal3-deploy/0.1.0/charts/ironic/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  dns_provider: cloudflare
+  cloudflare:
+    apiToken: "foo"
+    proxied: false
+  ```
+
+Ironic helm-chart values.yaml or overrides file
+  ```
+  global.dnsDomain: baremetal.management ( Valid Domain Name registered in CloudFlare)
+  baremetaloperator.cloudflareApiToken: "foo" (Valid cloudflare Api Token for the domain registered)
```
Comment (e-minguez) [inline charts/metal3-deploy/0.1.0/charts/ironic/README.md:33]:
```suggestion
global.dnsDomain: baremetal.management (Valid Domain Name registered in CloudFlare)
baremetaloperator.cloudflareApiToken: "foo" (Valid CloudFlare API Token for the domain registered)
```

**Review by e-minguez — COMMENTED** [review]
Some small nits.

File: .gitignore [hunk 6]
```diff
@@ -1,4 +1,4 @@
-bin
+/bin/
```
[the remark this answers is not in the document]
Comment (dbw7) [reply .gitignore:1]: This one was actually a bit of a funny bug: We have two bin folders inside of the the entire charts repo, one of them is meant to be ignored but the other one is necessary. "bin" ignores both of them (which is why I was so confused cause I thought I pushed it but it kept disappearing 🤣) but /bin/ only ignores the unnecessary one 
image

[the remark this answers is not in the document]
Comment (e-minguez) [reply .gitignore:1]: what about `./bin` instead?

[the remark this answers is not in the document]
Comment (dbw7) [reply .gitignore:1]: That brings back the unecessary one
image

**Pull request merged**